File: SrdDatabase/Models/Sacco/Members/Member.cs
using System;

namespace SrdDatabase.Models.Sacco.Members
{
    public class Member
    {
        public int Id { get; }

        public int AccountNumber { get; }

        public string Name { get; }

        public bool IsChurch { get; }

        public DateTime DateJoined { get; }

        public DateTime AutoFeesStartDate { get; }

        public int YearsOfFees { get; }

        public int Shares { get; }

        public int SharesValue { get; }

        public int Savings { get; }

        public int Balance { get; }

        public Member(
            int id,
            int accountNumber,
            string name,
            bool isChurch,
            DateTime dateJoined,
            DateTime autoFeesStartDate,
            int yearsOfFees,
            int shares,
            int sharesValue,
            int savings,
            int balance)
        {
            Id = id;
            AccountNumber = accountNumber;
            Name = name;
            IsChurch = isChurch;
            DateJoined = dateJoined;
            AutoFeesStartDate = autoFeesStartDate;
            YearsOfFees = yearsOfFees;
            Shares = shares;
            SharesValue = sharesValue;
            Savings = savings;
            Balance = balance;
        }

        // for Dapper
        public Member()
        {
        }
    }
}

File: SrdDatabase/ClientApp/src/models/sacco/member.ts
import { PagedParameters, PagedResults } from "../shared";
import { DistributionAppliedResults } from "./distribution";
import { LoanResults } from "./loan";
import { TransactionResults } from "./transaction";
import { PaymentResults } from "./payment";

export interface Member {
    id?: number;
    accountNumber?: number;
    name?: string;
    isChurch?: boolean;
    dateJoined?: Date;
    autoFeesStartDate?: Date;
    yearsOfFees?: number;
    shares?: number;
    sharesValue?: number;
    savings?: number;
    balance?: number;
}

export interface MemberToSend extends Omit<Member, "dateJoined"> {
    dateJoined?: string;
}

export interface MemberDetails {
    member: Member;
    transactionResults: TransactionResults;
    distributionAppliedResults: DistributionAppliedResults;
    loanResults: LoanResults;
    paymentResults: PaymentResults;
}   

export interface MemberParameters extends PagedParameters {
    accountNumber?: number;
    name?: string;
    isChurch?: boolean;
}

export interface MemberResults extends PagedResults {
    members: Member[];
}